contracts/actions/flashloan/FLAaveV3CarryDebt.sol

```diff
@@ -0,0 +1,151 @@
+                ? reserveData.variableDebtTokenAddress 
+                : reserveData.stableDebtTokenAddress;
+            
+            uint256 creditDelegationAllowance = IDebtToken(debtToken).borrowAllowance(
+                flData.onBehalfOf,
+                address(this)
+            );
+            
+            if (creditDelegationAllowance > 0) {
+                revert CreditDelegationAllowanceLeftError(creditDelegationAllowance);
+            }
+        }
+
+        return bytes32(flAmount);
+    }
+
+    // solhint-disable-next-line no-empty-blocks
+    function executeActionDirect(bytes memory _callData) public override payable {}
+
+    /// @inheritdoc ActionBase
+    function actionType() public override pure returns (uint8) {
+        return uint8(ActionType.FL_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Gets a Fl from AaveV3 and returns back the execution to the action address
+    /// @param _flData All the amounts/tokens and related aave fl data
+    /// @param _params Rest of the data we have in the recipe
+    function _flAaveV3(FlashLoanParams memory _flData, bytes memory _params) internal returns (uint) {
+
+        IPoolV3(AAVE_V3_LENDING_POOL).flashLoan(
+            address(this),
+            _flData.tokens,
+            _flData.amounts,
+            _flData.modes,
+            _flData.onBehalfOf,
+            _params,
+            AAVE_REFERRAL_CODE
+        );
+
+        emit ActionEvent(
+            "FLAaveV3CarryDebt",
+            abi.encode(_flData.tokens, _flData.amounts, _flData.modes, _flData.onBehalfOf)
+        );
+
+        return _flData.amounts[0];
+    }
+
+    /// @notice Aave callback function that formats and calls back RecipeExecutor
+    /// @dev FL amount is not returned, instead debt position is opened
+    function executeOperation(
+        address[] memory _assets,
+        uint256[] memory _amounts,
+        uint256[] memory _fees,
+        address _initiator,
+        bytes memory _params
+    ) public nonReentrant returns (bool) {
+        if (msg.sender != AAVE_V3_LENDING_POOL){
+            revert OnlyAaveCallerError();
+        }
+        if (_initiator != address(this)){
+            revert SameCallerError();
+        }
+
+        (Recipe memory currRecipe, address proxy) = abi.decode(_params, (Recipe, address));
+
+        // Send FL amounts to user proxy
+        for (uint256 i = 0; i < _assets.length; ++i) {
+            _assets[i].withdrawTokens(proxy, _amounts[i]);
+        }
+
+        address payable recipeExecutor = payable(registry.getAddr(RECIPE_EXECUTOR_ID));
+
+        // call Action execution
+        IDSProxy(proxy).execute{value: address(this).balance}(
+            recipeExecutor,
+            abi.encodeWithSignature("_executeActionsFromFL((string,bytes[],bytes32[],bytes4[],uint8[][]),bytes32)", currRecipe, bytes32(_amounts[0] + _fees[0]))
```

Comment:
this action will never actually take fees, even though _fees[0] will be different from 0
https://github.com/aave/aave-v3-core/blob/6070e82d962d9b12835c88e68210d0e63f08d035/contracts/protocol/libraries/logic/FlashLoanLogic.sol#L158

this should be just _amounts[0], so it can be later piped to delegate credit action inside the recipe with $1